src/components/SubscriptionManagement/SubscriptionManagement.js

```diff
@@ -26,6 +31,11 @@ const SubscriptionManagement = ({ subscription, showMessageBox }) => {
   const { data: switchSettings } = useSelector(
     store => store.plan.switchSettings
   );
+
+  const { isLoading: isRetentionActionsLoading } = useSelector(
+    selectRetentionActions
```

Comment:
we could create selectRetentionActionsIsLoading selector to avoid unnescessary rerenders and destructuring.